From e3147045513c7cffdf3b8b124b9bcc118efcd662 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Wed, 15 Sep 2010 14:54:17 -0400
Subject: [PATCH] Don't show shutdown options when not available

The logic for showing the shut down menu was a little
iffy before.  This is apparently causing issues on solaris,
and probably elsewhere.

https://bugzilla.gnome.org/show_bug.cgi?id=624298
(cherry picked from commit d8a75a806215cf95832de70da7fdb9c7592d697f)
---
 gui/simple-greeter/gdm-greeter-panel.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gui/simple-greeter/gdm-greeter-panel.c b/gui/simple-greeter/gdm-greeter-panel.c
index 06563dc..5cb2ef2 100644
--- a/gui/simple-greeter/gdm-greeter-panel.c
+++ b/gui/simple-greeter/gdm-greeter-panel.c
@@ -855,7 +855,7 @@ setup_panel (GdmGreeterPanel *panel)
                 gtk_widget_show (panel->priv->hostname_label);
         }
 
-        if (panel->priv->display_is_local || get_show_restart_buttons (panel)) {
+        if (!panel->priv->display_is_local || get_show_restart_buttons (panel)) {
                 GtkWidget *menu_item;
                 GtkWidget *image;
 
@@ -882,7 +882,7 @@ setup_panel (GdmGreeterPanel *panel)
                         menu_item = gtk_menu_item_new_with_label ("Disconnect");
                         g_signal_connect (G_OBJECT (menu_item), "activate", G_CALLBACK (do_disconnect), NULL);
                         gtk_menu_shell_append (GTK_MENU_SHELL (panel->priv->shutdown_menu), menu_item);
-                } else {
+                } else if (get_show_restart_buttons (panel)) {
                         if (can_suspend ()) {
                                 menu_item = gtk_menu_item_new_with_label (_("Suspend"));
                                 g_signal_connect (G_OBJECT (menu_item), "activate", G_CALLBACK (do_system_suspend), NULL);
-- 
1.7.1

